### PR TITLE
Search bar

### PR DESCRIPTION
**Pull request by AngelaLee18**
MVC

SearchBar and pickerView

DESCRIPTION

In this branch I created the searchBar and pickerView to perform searches according to the selected picker and what was entered in the search bar. And in the play list is only showed the result about the according to all the filter. 

EVIDENCE

https://user-images.githubusercontent.com/109097550/187936604-f2920427-4d3b-49a1-95c4-c370d6dfbf45.mov

### CODE REVIEW

File: Ted Talk/Ted Talk/View/SearchBarCellModel.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    init(SearchInfo: TedTalkData) {
+        event = SearchInfo.event
+        main_speaker = SearchInfo.main_speaker
+        title = SearchInfo.title
+        name = SearchInfo.name
+        description = SearchInfo.description
+    }
+    
+    
+}
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/View/SearchBarCellModel.swift]: For what is that class?

File: Ted Talk/Ted Talk/Controller/ViewController.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        return result
+    }
+
+}
+
+
+
+
+
+
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Controller/ViewController.swift]: do not add extra lines.

File: Ted Talk/Ted Talk/Controller/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    }
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        tableViewData = callSearchenteredWord()
+    }
+    
+    func callSearchenteredWord() -> [TedTalkData] {
+        var result: [TedTalkData]
+        result = manager.searchEnteredWord(searchText: searchBar.text ?? "", picker: pickerOptions[pickerView.selectedRow(inComponent: 0)])
+        return result
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Controller/ViewController.swift]: you don't need this result array. Just return manager.searchEnteredWord(...

File: Ted Talk/Ted Talk/Model/DataManager.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+//
+//  Created by Angela Lee on 25/08/2022.
+//
+
+import Foundation
+
+public class DataManager {
+    var fileName: String
+    var tedTalks: [TedTalkData] = []
+    func getDataTedTalks(completionHandler: @escaping ([TedTalkData]) -> Void) {
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Model/DataManager.swift]: Add space between properties and func, y also recomend to use marks: https://cocoacasts.com/organize-your-swift-project-with-annotations-todos-and-fixmes

File: Ted Talk/Ted Talk/Model/DataManager.swift [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                return talk.main_speaker.lowercased().contains(searchText.lowercased())
+            case "Title":
+                return talk.title.lowercased().contains(searchText.lowercased())
+            case "Name":
+                return talk.name.lowercased().contains(searchText.lowercased())
+            case "Description":
+                return talk.description.lowercased().contains(searchText.lowercased())
+                
+            default:
+                return talk.event.lowercased().contains(searchText.lowercased())
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Model/DataManager.swift]: this is wrong, the default case should be "All"

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]
The PR should point to Module_2/main branch!

File: Ted Talk/Ted Talk/View/SearchBarCellModel.swift (hunk 1 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/View/SearchBarCellModel.swift]: e291d24a351e3b4f108b2f0c998ea0711b95c960

File: Ted Talk/Ted Talk/Controller/ViewController.swift (hunk 2 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Controller/ViewController.swift]: 8bf510de592cf4a3430bad7daa807e3305c18cfb

File: Ted Talk/Ted Talk/Controller/ViewController.swift (hunk 3 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Controller/ViewController.swift]: fd09155f3d4178703777a575543d0ca86c279d3f

File: Ted Talk/Ted Talk/Model/DataManager.swift (hunk 5 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Model/DataManager.swift]: b5c3a9ba7a86befc08ae71810a70f705f706bc3a

File: Ted Talk/Ted Talk/Model/DataManager.swift (hunk 4 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Model/DataManager.swift]: b5c3a9ba7a86befc08ae71810a70f705f706bc3a

File: Ted Talk/Ted Talk/Model/DataManager.swift (hunk 5 above)
Comment (AngelaLee18) [reply Ted Talk/Ted Talk/Model/DataManager.swift]: 54b6ddb6f18c698e7ce79d9b875c384d7e74cc02

**Comment by ezequielvelezglobant** [comment]
@AngelaLee18 Please review the UI, there is an extra table that is not needed. And the table should start right before the pickerView.
image

**Review by ezequielvelezglobant — APPROVED** [review]

**Review by JMVarelaGlobant — APPROVED** [review]

**Pull request merged**